charms/reactive/decorators.py

```diff
@@ -222,3 +223,25 @@ def only_once(action=None):
     handler.add_predicate(lambda: not was_invoked(action_id))
     handler.add_post_callback(partial(mark_invoked, action_id))
     return action
+
+
+def collect_metrics():
+    """
+    Register the decorated function to run for the collect_metrics hook.
+    """
```

Comment:
If you reused the existing hook phase, these two decorators would not be needed. You could just use `@hook('collect_metrics')`